src/plugins/messenger/MessengerPreview/components/MessengerGenericTemplate/MessengerGenericTemplate.tsx

```diff
@@ -86,25 +87,55 @@ export const getMessengerGenericTemplate = ({
         IMessengerGenericTemplateState
         > {
 
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                selectedItem: 0,
+            }
+        }
+
+        // Change the selectedItem state, in order to scroll the card with a focused element into view
         handleScrollToView = (index) => {
-            const focusedButton = document.activeElement;
-            focusedButton?.scrollIntoView({ behavior: "smooth", inline: "center", block: "nearest"});
+            this.setState({selectedItem: index})
+        }
+
+        handleCardChange = (index) => {
+            this.setState({selectedItem: index});
+        }
+
+        /**
+         * Move cards in carousel using arrow keys. 'useKeyboardArrows' prop of react-responsive-carousel
+         * moves the carousel irrespective of it has a focused item or not. Therefore, it has to be handled this way.
+         * 
+         */
+        handleArrowKeyDown = (event) => {
+            if(event.key === "ArrowRight") {
+                this.setState({selectedItem: this.state.selectedItem + 1})
+            } else if(event.key === "ArrowLeft") {
+                this.setState({selectedItem: this.state.selectedItem - 1})
+            }
         }
 
         renderElement = (element: IFBMGenericTemplateElement, index?: number) => {
             const { onAction, ...divProps } = this.props;
             const { image_url, image_alt_text, title, subtitle, buttons, default_action } = element;
 
+            const carouselListLength = this.props.payload.elements.length;
             const isCentered = this.props.config.settings.designTemplate === 2;
             const carouselRootId = `webchatCarouselTemplateRoot-${uuid.v4()}`;
             const carouselTitleId = `webchatCarouselTemplateTitle-${uuid.v4()}`;
             const carouselSubtitleId = `webchatCarouselTemplateSubtitle-${uuid.v4()}`;
+
+            const listItemCount = index !== undefined ? `${index + 1} of ${carouselListLength}` : undefined;
+            const carouselAriaLabel = title ? `${listItemCount}: ${title}` : listItemCount;
             const carouselAriaLabelledby = title ? carouselTitleId : undefined;
             const carouselAriaDescribedby = subtitle ? carouselSubtitleId : undefined;
-            const a11yProps = {role: "group", "aria-labelledby": carouselAriaLabelledby, "aria-describedby": carouselAriaDescribedby};
+            const carouselRooTA11yProps = {role: "group", "aria-labelledby": listItemCount ? undefined : carouselAriaLabelledby,
```

Comment:
is this spelled correct? Or should the T be in lower case?

Reply:
Thanks for pointing these out. I have pushed changes to fix those. 
1. You will not see the focused (grey background) style for elements that are not links
2. When a carousel element is brought into view using Arrow keys, you will now be able to focus the first interactive element within it, on clicking 'Tab'. 
3. Also, fixed the typo.